bootloader/lk.go

```diff
@@ -49,8 +60,18 @@ func newLk(rootdir string, opts *Options) Bootloader {
 		//      are very different from runtime vs image-building mode.
 		//
 		// determine mode we are in, runtime or image build
-		l.inRuntimeMode = !opts.PrepareImageTime
+
+		l.prepareImageTime = opts.PrepareImageTime
+
+		l.role = opts.Role
```

Comment:
if feels like somewhere we should sanity check that if PrepareImageTime then Role is Sole or Recovery. I suppose it should always be the case so it could go in validate() ?

Reply:
I opened a different PR for this: #9746